### PR TITLE
Fix .push() not considering external functions

### LINKED ISSUE #12546: [Sol->Yul] ICE due to invalid IR when compiling push on dynamic storage array of one of it's own elements
```
contract C {
  struct St1 {
    function () external el;
  }
  St1[] s;
  function f() public {
    s.push(s[0]);
  }
}
```

throws

https://github.com/ethereum/solidity/blob/07d1bc5f37ab071ec8a33d98f663462ce83c38ea/libsolidity/codegen/ir/IRGenerator.cpp#L112

Repro

```
$ solc --ir test.sol
```

### PR DESCRIPTION
**Pull request by Marenz**
fixes #12546

### CODE REVIEW

**Comment by ekpyron** [comment]
It's a bit silly to split the function pointer into two slots only to then recombine them to a single one instead of just ``sload`` and ``sstore`` the same value...  but it's also an advantage to keep the code generic and simple, so it may even better to just keep it like this - I'm wondering how well this optimizes :-).

**Comment by ekpyron** [comment]
I'm also wondering if we should be more paranoid about testing this in various corner cases...

**Review by ekpyron — COMMENTED** [review]
Well, at least a Changelog entry would be good, apart from that I may approve it as is :-).

File: Changelog.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Compiler Features:
 Bugfixes:
  * Antlr Grammar: Allow builtin names in ``yulPath`` to support ``.address`` in function pointers.
  * Control Flow Graph: Perform proper virtual lookup for modifiers for uninitialized variable and unreachable code analysis.
+ * IR Generator: Fix IR syntax error when copying storage arrays of structs containing functions.
  * Immutables: Fix wrong error when the constructor of a base contract uses ``return`` and the parent contract contains immutable variables.
```
Comment (ekpyron) [inline Changelog.md]: Do we sort this in case-sensitive or case-insensitive manner :-D?

**Review by ekpyron — DISMISSED** [review]

File: Changelog.md (hunk 1 above)
Comment (Marenz) [reply Changelog.md]: well, I used `:sort` in vim :> I can change it

Comment (ekpyron) [reply Changelog.md]: I'm not saying we should order it differently, just that I'm not sure, if we have a fixed convention for this :-).

**Review by ekpyron — APPROVED** [review]

**Pull request merged**